mon-pix/app/controllers/campaigns/restricted/join.js

```diff
@@ -148,6 +152,26 @@ export default class JoinRestrictedCampaignController extends Controller {
     });
   }
 
+  _showErrorMessageByCode(meta) {
+    const ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION = `Vous possédez déjà un compte Pix avec l’adresse e-mail ${meta.value}<br> Pour continuer, connectez-vous à ce compte ou demandez de l’aide à un enseignant.<br>(Code R11)`;
+    const ACCOUNT_WITH_USERNAME_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION = `Vous possédez déjà un compte Pix utilisé avec l’identifiant ${meta.value}<br> Pour continuer, connectez-vous à ce compte ou demandez de l'aide à un enseignant.<br>(Code R12)`;
+    const ACCOUNT_WITH_GAR_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION = 'Vous possédez déjà un compte Pix via l\'ENT dans un autre établissement scolaire.<br> Pour continuer, contactez un enseignant qui pourra vous donner l’accès à ce compte à l\'aide de Pix Orga.';
+    const ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_SAME_ORGANIZATION = `Vous possédez déjà un compte Pix utilisé dans votre établissement scolaire, avec l'adresse mail ${meta.value}.<br> Connectez-vous à ce compte sinon demandez de l'aide à un enseignant. (Code R31)`;
+    const ACCOUNT_WITH_USERNAME_ALREADY_EXIST_FOR_THE_SAME_ORGANIZATION = `Vous possédez déjà un compte Pix utilisé dans votre établissement scolaire, avec un identifiant sous la forme ${meta.value}.<br> Connectez-vous à ce compte sinon demandez de l'aide à un enseignant. (Code R32)`;
+    const ACCOUNT_WITH_GAR_ALREADY_EXIST_FOR_THE_SAME_ORGANIZATION = 'Vous possédez déjà un compte Pix via l\'ENT dans votre établissement scolaire.<br> Connectez-vous à ce compte sinon demandez de l\'aide à un enseignant. (Code R33)';
+
+    const errorsMessagesByCodes = {
+      'R11': ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION,
+      'R12': ACCOUNT_WITH_USERNAME_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION,
+      'R13': ACCOUNT_WITH_GAR_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION,
+      'R31': ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_SAME_ORGANIZATION,
+      'R32': ACCOUNT_WITH_USERNAME_ALREADY_EXIST_FOR_THE_SAME_ORGANIZATION,
+      'R33': ACCOUNT_WITH_GAR_ALREADY_EXIST_FOR_THE_SAME_ORGANIZATION,
+      'default': this.intl.t(ENV.APP.API_ERROR_MESSAGES.INTERNAL_SERVER_ERROR.MESSAGE),
+    };
+    return (errorsMessagesByCodes[meta.shortCode] || errorsMessagesByCodes['default']);
+  }
+
```

Comment:
Je propose de renommer la méthode en `_showErrorMesssageByShortCode(meta)`
Je vois bien aussi quelque chose dans ce style : 
```suggestion
  _showErrorMessageByShortCode(meta) {
    const default = this.intl.t(APP.API_ERROR_MESSAGES.INTERNAL_SERVER_ERROR.MESSAGE)
    const errors = [
      { 
      code: 'ACCOUNT_WITH_EMAIL_ALREADY_EXIST_FOR_ANOTHER_ORGANIZATION ', 
      shortCode: 'R11', 
      message: `Vous possédez déjà un compte Pix utilisé avec l’identifiant ${meta.value}<br> Pour continuer, connectez-vous à ce compte ou demandez de l'aide à un enseignant.<br>(Code R12)` 
      }
    ]

    return _.find(errors, ['shortCode', meta.shortCode]) || default; 
  }
```

Reply:
Done thanks